ubuntu_image/__main__.py

```diff
@@ -158,15 +186,106 @@ def parseargs(argv=None):
+            'snap',
+            help=_("""Create snap-based Ubuntu Core image."""))
+    classic_cmd = subparser.add_parser(
+            'classic',
+            help=_("""Create debian-based Ubuntu Classic image."""))
+    argv = get_modified_args(subparser, 'snap', argv)
+
+    snap_cmd = add_common_args(snap_cmd)
+    classic_cmd = add_common_args(classic_cmd)
+
+    # Snap-based image options.
+    snap_cmd.add_argument(
+        'model_assertion', nargs='?',
+        help=_("""Path to the model assertion file.  This argument must be
+        given unless the state machine is being resumed, in which case it
+        cannot be given."""))
+    snap_cmd.add_argument(
+        '--extra-snaps',
+        default=None, action='append',
+        help=_("""Extra snaps to install.  This is passed through to `snap
+        prepare-image`."""))
+    snap_cmd.add_argument(
+        '--cloud-init',
+        default=None, metavar='USER-DATA-FILE',
+        help=_('cloud-config data to be copied to the image'))
+    snap_cmd.add_argument(
+        '-c', '--channel',
+        default=None,
+        help=_('The snap channel to use'))
+
+    # Classic-based image options.
+    classic_cmd.add_argument(
+        'gadget_tree', nargs='?',
+        help=_("""Gadget tree.  This is a tree equivalent to an unpacked
+        gadget snap at core image build time."""))
+    classic_cmd.add_argument(
+        '-p', '--project',
+        default='ubuntu-cpc', metavar='PROJECT',
+        help=_("""Project name to be specified to livecd-rootfs."""))
+    classic_cmd.add_argument(
+        '-s', '--suite',
+        default=get_host_distro(), metavar='SUITE',
+        help=_("""Distribution name to be specified to livecd-rootfs."""))
+    classic_cmd.add_argument(
+        '-a', '--arch',
+        default=get_host_arch(), metavar='CPU-ARCHITECTURE',
+        help=_("""CPU architecture to be specified to livecd-rootfs.
+        default value is builder arch."""))
+    classic_cmd.add_argument(
+        '--subproject',
+        default=None, metavar='SUBPROJECT',
+        help=_("""Sub project name to be specified to livecd-rootfs."""))
+    classic_cmd.add_argument(
+        '--subarch',
+        default=None, metavar='SUBARCH',
+        help=_("""Sub architecture to be specified to livecd-rootfs."""))
+    classic_cmd.add_argument(
+        '--with-proposed',
+        default=False, action='store_true',
+        help=_("""Proposed repo to install, This is passed through to
+        livecd-rootfs."""))
+    classic_cmd.add_argument(
+        '--image-format',
+        default='img',
+        help=_("""Image format to be specified to livecd-rootfs."""))
+    classic_cmd.add_argument(
+        '--extra-ppas',
+        default=None, action='append',
+        help=_("""Extra ppas to install. This is passed through to
+        livecd-rootfs."""))
+
     args = parser.parse_args(argv)
+
     if args.debug:
         logging.basicConfig(level=logging.DEBUG)
+
```

Comment:
I'd prefer if we'd stick to the original coding-style for ubuntu-image, i.e. not introducing any unneeded newlines. But this is cosmetics.